tensorflowonspark/TFSparkNode.py

```diff
@@ -295,20 +298,26 @@ def _mapfn(iter):
       logger.info("export TF_CONFIG: {}".format(tf_config))
       os.environ['TF_CONFIG'] = tf_config
 
-    # reserve GPU(s) again, just before launching TF process (in case situation has changed)
-    if compat.is_gpu_available():
-      # compute my index relative to other nodes on the same host (for GPU allocation)
-      my_addr = cluster_spec[job_name][task_index]
-      my_host = my_addr.split(':')[0]
-      flattened = [v for sublist in cluster_spec.values() for v in sublist]
-      local_peers = [p for p in flattened if p.startswith(my_host)]
-      my_index = local_peers.index(my_addr)
-
-      num_gpus = tf_args.num_gpus if 'num_gpus' in tf_args else 1
-      gpus_to_use = gpu_info.get_gpus(num_gpus, my_index)
-      gpu_str = "GPUs" if num_gpus > 1 else "GPU"
-      logger.debug("Requested {} {}, setting CUDA_VISIBLE_DEVICES={}".format(num_gpus, gpu_str, gpus_to_use))
-      os.environ['CUDA_VISIBLE_DEVICES'] = gpus_to_use
+    if version.parse(pyspark.__version__) >= version.parse("3.0.0"):
```

Comment:
also I'm not sure how particular you want to be here, you could make this check:
    if version.parse(pyspark.__version__).base_version >= version.parse("3.0.0").base_version:

and it would work with the spark 3.0 snapshot/dev versions you can build now.

Reply:
Thanks for taking a look at this!  Are you aware of any Spark APIs which might indicate if GPU scheduling is available or not (besides just falling back on empty/fail)?

Yes, I can use `base_version` for now on this branch... but likely will only merge this PR when Spark 3 is officially released.

Reply:
no there isn't an api if its enabled because really its generate to any resource type. Right now you could check the spark confs to see if its set, but that won't be true when stage level schedule is merged. if the user enables GPU scheduling the executor won't start unless it has the number of gpus requested by the user, so its pretty safe to assume if that is empty they didn't enable it.